### Exploratory Data Analysis

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

In [3]:
# Load dropout-ratio dataset from data folder
dropout_df = pd.read_csv("../data/dropout-ratio-2012-2015.csv")

In [6]:
# Explore the top five rows
dropout_df.head()

State_UT     year Primary_Boys Primary_Girls Primary_Total  \
0   A & N Islands  2012-13         0.83          0.51          0.68   
1   A & N Islands  2013-14         1.35          1.06          1.21   
2   A & N Islands  2014-15         0.47          0.55          0.51   
3  Andhra Pradesh  2012-13          3.3          3.05          3.18   
4  Andhra Pradesh  2013-14         4.31          4.39          4.35   

  Upper Primary_Boys Upper Primary_Girls Upper Primary_Total Secondary _Boys  \
0     Uppe_r_Primary                1.09                1.23            5.57   
1                 NR                1.54                0.51            8.36   
2               1.44                1.95                1.69           11.47   
3               3.21                3.51                3.36           12.21   
4               3.46                4.12                3.78           11.95   

  Secondary _Girls Secondary _Total HrSecondary_Boys HrSecondary_Girls  \
0             5.55             5.56            17.66             10.15   
1             5.98              7.2            18.94              12.2   
2             8.16             9.87            21.05             12.21   
3            13.25            12.72             2.66                NR   
4            13.37            12.65            12.65             10.85   

  HrSecondary_Total  
0             14.14  
1             15.87  
2             16.93  
3              0.35  
4             11.79

In [9]:
# Check how many rows and columns using shape
dropout_df.shape

(110, 14)

#### We have 110 rows and 14 columns of data point in the dataset

In [11]:
# Statistics of all the dataset
dropout_df.describe(include="all")

State_UT     year Primary_Boys Primary_Girls Primary_Total  \
count       110      110          110           110           110   
unique       40        3           93            87            87   
top     Haryana  2013-14           NR            NR            NR   
freq          3       37           15            15            17   

       Upper Primary_Boys Upper Primary_Girls Upper Primary_Total  \
count                 110                 110                 110   
unique                 91                  97                  96   
top                    NR                  NR                  NR   
freq                   16                   8                   9   

       Secondary _Boys Secondary _Girls Secondary _Total HrSecondary_Boys  \
count              110              110              110              110   
unique             106              107              106               81   
top                 NR               NR               NR               NR   
freq                 4                4                4               28   

       HrSecondary_Girls HrSecondary_Total  
count                110               110  
unique                76                78  
top                   NR                NR  
freq                  34                31

In [12]:
# Check missing values
dropout_df.isnull().sum()

State_UT               0
year                   0
Primary_Boys           0
Primary_Girls          0
Primary_Total          0
Upper Primary_Boys     0
Upper Primary_Girls    0
Upper Primary_Total    0
Secondary _Boys        0
Secondary _Girls       0
Secondary _Total       0
HrSecondary_Boys       0
HrSecondary_Girls      0
HrSecondary_Total      0
dtype: int64

#### As showed above there are no missing values accross all columns and rows of the data point in the dataset.

In [13]:
# Check out the types of data in the dataset
dropout_df.dtypes

State_UT               object
year                   object
Primary_Boys           object
Primary_Girls          object
Primary_Total          object
Upper Primary_Boys     object
Upper Primary_Girls    object
Upper Primary_Total    object
Secondary _Boys        object
Secondary _Girls       object
Secondary _Total       object
HrSecondary_Boys       object
HrSecondary_Girls      object
HrSecondary_Total      object
dtype: object